proxy/http2/Http2ConnectionState.cc

```diff
@@ -1007,14 +1007,18 @@ Http2ConnectionState::send_headers_frame(FetchSM *fetch_sm)
 
   DebugHttp2Stream(ua_session, stream->get_id(), "Send HEADERS frame");
 
-  http2_convert_header_from_1_1_to_2(resp_header);
-  buf_len = resp_header->length_get() * 2; // Make it double just in case
+  HTTPHdr *h2_hdr = http2_generate_h2_header_from_1_1(resp_header);
```

Comment:
I think we should probably avoid the malloc by using an out parameter for the HTTP2 copy.